### DISCUSSION TITLE
Use CAN but not for NMEA2000

### DISCUSSION THREAD

**Discussion by jakobwal** [hatlabs/discussions#29] [Q&A] [answered]
Hi

I want to use the CAN interface but not connected to an NMEA2000 bus. I struggle with understanding how to connect CAN V+. The device I want to interface specifies +4.6V on its CAN interface. Unsure if this is output or input...? On the SH-ESP32 its specified +12V but when I read the datasheet for the CAN transceiver ISO1050DUB the supply voltage is 5V.

If I power the SH-ESP32 through the standard power input terminal, will that also power the CAN transceiver or do I need to supply +12V on the CAN pin?
Can I get +5V from the SH-ESP32 to power my external CAN device? For example from the USB 5V pin? Is this powered even when no USB device is connected?

Am I going about this the right way? :-)

**Comment by mairas** [comment, marked as answer]
You don't have to worry about the CANH and CANL voltages. They're provided by the CAN transceiver, and as long as the transceiver is compliant with the standard (ISO 11898-2), the signal voltages are all right.

SH-ESP32 implements an isolated CAN bus interface, meaning that the ground and supply voltage of the CAN interface are (at least by default) completely separated from the rest of the board. The interface is powered using the NMEA 2000 12V feed and then converted to 3.3V with an LDO regulator.

(Note that from v2.0.0 onwards, a separate digital isolator and an unisolated CAN transceiver is used instead of the ISO1050 which performs both functions in one package. This is due to better part availability. The SN65HVD23x transceivers work at 3.3V instead of 5V required by ISO1050. I just added the 2.0.1 schematics at https://hatlabs.github.io/sh-esp32/pages/revisions/#version-201.)

Soo, anyhow, you're right that the transceiver needs to be powered somehow. If you feed 12V (anything between 5-32V should work) and GND to the CAN input, that'll do. The USB 5V is only provided when the USB cable is connected, so that's no good, but since the new transceiver works at 3.3V, you could connect the main 3.3V voltage (and GND) to CAN_3V3. I didn't anticipate that need and there are no dedicated pads for that, so soldering will require a bit of steady hand, unfortunately. With either option you lose the galvanic isolation, of course.

For the next major update, I think I should use an isolated DC-DC converter to provide the 3.3V voltage to the transceiver. That would simplify the design quite a bit and work better for non-N2K use cases.

**Reply by jakobwal** [reply]
Good stuff! I think I will just feed it with +12V and GND and loose galvanic isolation that way. The device I'm interfacing can also have its CAN transceiver powered internally so I will do that for that one as well.

With this setup, from what I understand, all I need to connect is CANH and CANL.

Thanks

**Reply by mairas** [reply]
Sounds good. Make sure that the devices share a common GND, though, especially if you work with mains-powered devices. If the ground is floating, data transmission still is likely to work but the common-mode voltage difference (the average voltage difference between CANH & CANL and GND) may become high enough to damage the transceiver. Been there, done that...

In short, best to connect CANH, CANL, and GND.

**Reply by jakobwal** [reply]
Thanks for the advice will do! And thanks in general for the excellent support here. I just ordered another SH-ESP32, this time with the new engine HAT. Now I just have to get everything to work :-)
